analyzer/src/test/java/com/novoda/sqlite/impl/SQLFileTest.java

```diff
@@ -0,0 +1,96 @@
+    private final static String STATEMENT_FOR_COMMENTS_WITHIN_TESTS_2 = "banana(_id INTEGER);";
+
+    @Test
+    public void givenSQLFile_whenParseAndGetStatements_thenGetCorrectStatements() throws IOException {
+        SQLFile file = givenSQLFileParsedFromString(STATEMENT_CREATE_TABLE + NEW_LINE + STATEMENT_ALTER_TABLE);
+
+        String[] actual = getStatementsFromFile(file);
+
+        assertArrayEquals(actual, EXPECTED_STATEMENTS);
+    }
+
+    @Test
+    public void givenSQLFileWithSpaces_whenParseAndGetStatements_thenGetCorrectTrimmedStatements() throws IOException {
+        SQLFile file = givenSQLFileParsedFromString(SPACE + STATEMENT_CREATE_TABLE + SPACE + NEW_LINE + SPACE + STATEMENT_ALTER_TABLE + SPACE);
+
+        String[] actual = getStatementsFromFile(file);
+
+        assertArrayEquals(actual, EXPECTED_STATEMENTS);
+    }
+
+    @Test
+    public void givenSQLFileWithLineComments_whenParseAndGetStatements_thenIgnoreComments() throws IOException {
+        SQLFile file = givenSQLFileParsedFromString(
+                STATEMENT_CREATE_TABLE + SPACE + LINE_COMMENT + NEW_LINE + STATEMENT_ALTER_TABLE + SPACE + LINE_COMMENT
+        );
+
+        String[] actual = getStatementsFromFile(file);
+
+        assertArrayEquals(actual, EXPECTED_STATEMENTS);
+    }
+
+    @Test
+    public void givenSQLFileWithBlockComments_whenParseAndGetStatements_thenIgnoreComments() throws IOException {
+        SQLFile file = givenSQLFileParsedFromString(
+                STATEMENT_CREATE_TABLE + NEW_LINE + BLOCK_COMMENT + NEW_LINE + STATEMENT_ALTER_TABLE);
+
+        String[] actual = getStatementsFromFile(file);
+
+        assertArrayEquals(actual, EXPECTED_STATEMENTS);
+    }
+
+    @Test
+    public void givenSQLFileWithLineCommentsWithinStatement_whenParseAndGetStatements_thenIgnoreComments() throws IOException {
+        SQLFile file = givenSQLFileParsedFromString(
+                STATEMENT_FOR_COMMENTS_WITHIN_TESTS_1 + NEW_LINE + LINE_COMMENT + NEW_LINE + STATEMENT_FOR_COMMENTS_WITHIN_TESTS_2
+        );
+
+        String[] actual = getStatementsFromFile(file);
+
+        assertEquals(actual.length, 1);
+        assertEquals(actual[0], STATEMENT_FOR_COMMENTS_WITHIN_TESTS_1 + SPACE + STATEMENT_FOR_COMMENTS_WITHIN_TESTS_2);
+    }
+
+    @Test
+    public void givenSQLFileWithBlockCommentsWithinStatement_whenParseAndGetStatements_thenIgnoreComments() throws IOException {
+        SQLFile file = givenSQLFileParsedFromString(
+                STATEMENT_FOR_COMMENTS_WITHIN_TESTS_1 + NEW_LINE + BLOCK_COMMENT + NEW_LINE + STATEMENT_FOR_COMMENTS_WITHIN_TESTS_2
+        );
+
+        String[] actual = getStatementsFromFile(file);
+
+        assertEquals(actual.length, 1);
+        assertEquals(actual[0], STATEMENT_FOR_COMMENTS_WITHIN_TESTS_1 + SPACE + STATEMENT_FOR_COMMENTS_WITHIN_TESTS_2);
+    }
+
+    private SQLFile givenSQLFileParsedFromString(String sql) throws IOException {
+        SQLFile file = new SQLFile();
```

Comment:
![ops, I did it again](http://i.giphy.com/FjWEXNradAzYY.gif)

aac255268a0ec3c3d10e9e4bcc08399c684878a4